File: src/server/bo/timeinterval/IllnessBO.py
from server.bo.timeinterval import TimeIntervalBO as ti

"""
@author Ha Mi Duong (https://github.com/HamiDuong)

IllnessBO ist eine Subklasse von TimeIntervallBO und stellt eine Arbeitspause eines Benutzers da
Weil TimeIntervalBO bereits von BusinessObject erbt, muss diese Klasse nicht nochmal importiert werden
"""
class IllnessBO (ti.TimeIntervalBO):

    """
    Konstruktor der Klasse IllnessBO
    geerbte Attribute                                   geerbt von
        _id                                             (BusinessObject -> TimeIntervalBO)
        _date_of_last_change                            (BusinessObject -> TimeIntervalBO)
        _start: Startpunkt des Zeitintervalls           (TimeIntervalBO)
        _end: Enpunkt des Zeitintervalls                (TimeIntervalBO)
        _start_event: Fremdschlüssel zum IllnessBeginBO (TimeIntervalBO)
        _end_event: Fremdschlüssel zum IllnessEndBO     (TimeIntervalBO)
        _type: Art des Intervalls - hier: "Illness"     (TimeIntervalBO)
    """
    def __init__(self):
        super().__init__()
        self.set_type("Illness")

    'Gibt die Werte eines Objekts der Klasse in Textform zurück'
    def __str__(self):
        return "Krankheit {}: von {} bis {}, Startevent: {}, Endevent: {}".format(self.get_id(), self.get_start(), self.get_end(), self.get_start_event(), self.get_end_event())

    'wandelt ein Python dict() in ein IllnessBO'
    def from_dict(dictionary=dict()):
        obj = IllnessBO()
        obj.set_id(dictionary["id"])
        obj.set_date_of_last_change(dictionary["dateOfLastChange"])
        obj.set_start(dictionary["start"])
        obj.set_end(dictionary["end"])
        obj.set_start_event(dictionary["startEvent"])
        obj.set_end_event(dictionary["endEvent"])
        obj.set_type(dictionary["type"])
        return obj

File: frontend/worktimeapp/src/API/IllnessBO.js
import BusinessObject from "./BusinessObject";

/**
 * @author Ha Mi Duong (https://github.com/HamiDuong)
 * 
 * Objekt für Kranktage
 */
export default class IllnessBO extends BusinessObject{
    constructor(start, end, startEvent, endEvent, type){
        super();
        this.start = start;
        this.end = end;
        this.startEvent = startEvent;
        this.endEvent = endEvent;
        this.type = type;
    }

    // Getter und Setter
    setStart(start){
        this.start = start;
    }

    getStart(){
        return this.start;
    }

    setEnd(end){
        this.end = end;
    }

    getEnd(){
        return this.end;
    }

    setStartEvent(sEvent){
        this.startEvent = sEvent;
    }

    getStartEvent(){
        return this.startEvent;
    }

    setEndEvent(eEvent){
        this.endEvent = eEvent;
    }

    getEndEvent(){
        return this.endEvent;
    }
    
    setType(type){
        this.type = type;
    }

    getType(){
        return this.type;
    }

    // wandelt JSON in IllnessBO
    static fromJSON(illness){
        let res = [];
        if(Array.isArray(illness)){
            illness.forEach((elem) => {
                Object.setPrototypeOf(elem, IllnessBO.prototype);
                res.push(elem)
            })
        }else{
            let elem = illness;
            Object.setPrototypeOf(elem, IllnessBO.prototype);
            res.push(elem)
        }
        return res;
    }
}